hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/smtable/SoftwareModuleAddUpdateWindow.java

```diff
@@ -204,61 +187,50 @@ private void resetOldValues() {
-        /* action button layout (save & dicard) */
-        final HorizontalLayout buttonsLayout = new HorizontalLayout();
-        buttonsLayout.setSizeFull();
-        buttonsLayout.addComponents(saveSoftware, closeWindow);
-        buttonsLayout.setComponentAlignment(saveSoftware, Alignment.BOTTOM_LEFT);
-        buttonsLayout.setComponentAlignment(closeWindow, Alignment.BOTTOM_RIGHT);
-        buttonsLayout.addStyleName("window-style");
 
         final Label madatoryStarLabel = new Label("*");
         madatoryStarLabel.setStyleName("v-caption v-required-field-indicator");
         madatoryStarLabel.setWidth(null);
-        final HorizontalLayout hLayout = new HorizontalLayout();
-        hLayout.setSizeFull();
-        hLayout.addComponents(typeComboBox, madatoryStarLabel);
-        hLayout.setComponentAlignment(typeComboBox, Alignment.TOP_LEFT);
-        hLayout.setComponentAlignment(madatoryStarLabel, Alignment.TOP_RIGHT);
-        hLayout.setExpandRatio(typeComboBox, 0.8f);
 
         /*
          * The main layout of the window contains mandatory info, textboxes
          * (controller Id, name & description) and action buttons layout
          */
-        final VerticalLayout mainLayout = new VerticalLayout();
-        mainLayout.setSpacing(Boolean.TRUE);
-        mainLayout.addStyleName("lay-color");
-        mainLayout.addComponent(madatoryLabel);
-        mainLayout.setComponentAlignment(madatoryLabel, Alignment.MIDDLE_LEFT);
-        mainLayout.addComponent(hLayout);
-        mainLayout.setComponentAlignment(hLayout, Alignment.MIDDLE_LEFT);
-        mainLayout.addComponents(nameTextField, versionTextField, vendorTextField, descTextArea, buttonsLayout);
-        
+        addStyleName("lay-color");
+
+        final FormLayout formLayout = new FormLayout();
+        formLayout.addComponent(typeComboBox);
+        formLayout.addComponent(mandatoryLabel);
+        formLayout.addComponent(nameTextField);
+        formLayout.addComponent(versionTextField);
+        formLayout.addComponent(vendorTextField);
+        formLayout.addComponent(descTextArea);
+
+        setCompositionRoot(formLayout);
+
         /* add main layout to the window */
         window = SPUIComponentProvider.getWindow(i18n.get("upload.caption.add.new.swmodule"), null,
-                SPUIDefinitions.CREATE_UPDATE_WINDOW);
-        window.setContent(mainLayout);
-        window.setModal(true);
+                SPUIDefinitions.CREATE_UPDATE_WINDOW, this, event -> save(), event -> closeThisWindow(), null);
+        window.getButtonsLayout().removeStyleName("actionButtonsMargin");
         nameTextField.focus();
     }
 
     private void addDescriptionTextChangeListener() {
         descTextArea.addTextChangeListener(event -> {
             if (event.getText().equals(oldDescriptionValue) && vendorTextField.getValue().equals(oldVendorValue)) {
-                saveSoftware.setEnabled(false);
+                window.setSaveButtonEnabled(false);
```

Comment:
here you could set the enable direct:
`window.setSaveButtonEnabled(!(event.getText().equals(oldDescriptionValue) && vendorTextField.getValue().equals(oldVendorValue)))`